average.js

```diff
@@ -1,6 +1,7 @@
 
 function average(numbers) {
-  return numbers.reduce((p, c)=> p + c, 0) / numbers.length;
+  const sanitized =  numbers.filter(c => !isNaN(c));
+  return sanitized.reduce((p, c)=> p + c, 0) / sanitized.length;
 }
 
 module.exports = {average};
```

Comment:
Try a new test: The tests in `test/average.test.js` express the functionality of the `average` function.
The functionality accepts a sequence of readings from an IoT sensor. The sensor sends `NaN` when something goes wrong in measurement, which we have ignored.
Problem: If the sensor sends a lot of NaNs and very few readings, something is wrong. However, the `average` function ignores all the NaNs and still computes the average, as if nothing went wrong.
We need to add functionality to the average function to return `NaN`, when most of the input-numbers are `NaN`.
You need to:

- Reason out what 'most of the numbers' means: E.g., determine the number of NaNs in the input which would indicate that the sensor is unreliable. Mention any references you used to arrive at this logic
- Add a test that supplies such inputs and expects the output to be NaN
- Add code that satisfies this functionality, while not breaking existing functionality.

Reply:
I have defined a unique heuristic that determines whether the sensor is reliable or not

we can say that a sensor is unreliable if:
there is any sequence of NaN values longer than the square root of the input
or
there are more than 50% NaN values.

I have considered the case of sequence as if the sensor is not responding continuously, it will lead to insufficient data for that time period and the data wont be reliable.